modules/ctl/src/main/scala/command.scala

```diff
@@ -0,0 +1,149 @@
+package gem.ctl
+
+import net.bmjames.opts._
+import net.bmjames.opts.types.{ Success, Failure }
+import net.bmjames.opts.builder.internal.{ Mod, CommandFields }
+
+import scalaz._, Scalaz._, scalaz.effect._
+
+import gem.ctl.free.ctl.{ Server, Host }
+import gem.ctl.free.interpreter.Config
+
+/** A command to be interpreted by gemctl. */
+sealed trait Command extends Config
+
+/** Module of constructors of `Command`, and a parser too. */
+object Command {
+
+  case class Deploy(
+    server: Server,
+    deployRev:   String,
+    standalone:  Boolean,
+    verbose:     Boolean,
+    force:       Boolean
+  ) extends Command
+
+  case class Ps  (    server: Server, verbose: Boolean) extends Command
+  case class Stop(    server: Server, verbose: Boolean) extends Command
+  case class Log (    server: Server, verbose: Boolean, count: Int) extends Command
+  case class Rollback(server: Server, verbose: Boolean) extends Command
+
+  /**
+   * Construct a program to parse commandline `args` into a `Command`, or show help information if
+   * parsing fails. The `progName` argument is only used for help messages, and should match the
+   * name of the executable ("gemctl" for example).
+   */
+  def parse[A](progName: String, args: List[String]): IO[Option[Command]] =
+    execParserPure(prefs(idm[PrefsMod]), mainParser, args) match {
+      case Success(c) => IO(Some(c))
+      case Failure(f) =>
+        import Predef._
+        val (msg, exit) = renderFailure(f, progName)
+        IO.putStr(Console.BLUE) *>
+        IO.putStrLn(msg) *>
+        IO.putStrLn("")  *>
+        IO.putStrLn(s"""
+          |Hints:
+          |  $progName --help            To see available commands.
+          |  $progName COMMAND --help    To see help on a specific command.
+         """.trim.stripMargin) *>
+         IO.putStr(Console.RESET) as None
+    }
+
+  // Parser implementation below
+
+  private lazy val machine: Parser[Boolean] =
+    switch(
```

Comment:
Yeah it's from the optparse library. It constructs a "switch" like `--force` that's just an off/on argument.